### ISSUE TITLE
Class 'Laminas\Code\Annotation\AnnotationManager' not found

### ISSUE THREAD

**Issue by dbierer** [laminas/laminas-form#100]
### Bug Report

|    Q        |   A
|------------ | ------
| laminas/laminas-form | 2.16.2

#### Summary

Attempting to build a form using Laminas\Form\Annotation\Annotation\Builder

#### Current behavior

```
root@laminas [ /home/onlinemarket.complete ]# php annotation_test.php 
PHP Fatal error:  Uncaught Error: Class 'Laminas\Code\Annotation\AnnotationManager' not found in /home/onlinemarket.complete/vendor/laminas/laminas-form/src/Annotation/AnnotationBuilder.php:178
Stack trace:
#0 /home/onlinemarket.complete/vendor/laminas/laminas-form/src/Annotation/AnnotationBuilder.php(221): Laminas\Form\Annotation\AnnotationBuilder->getAnnotationManager()
#1 /home/onlinemarket.complete/vendor/laminas/laminas-form/src/Annotation/AnnotationBuilder.php(257): Laminas\Form\Annotation\AnnotationBuilder->getFormSpecification()
#2 /home/onlinemarket.complete/annotation_test.php(19): Laminas\Form\Annotation\AnnotationBuilder->createForm()
#3 {main}
  thrown in /home/onlinemarket.complete/vendor/laminas/laminas-form/src/Annotation/AnnotationBuilder.php on line 178
```

#### How to reproduce

1. `composer create-project -sdev laminas/laminas-mvc-skeleton path/to/install`
2. `cd path/to/install`
3. `composer require laminas/laminas-form`
4. `composer require laminas/laminas-code`
5. `php annotation_test.php`
```
<?php
// annotation_test.php
require __DIR__ . '/vendor/autoload.php';
use Laminas\Hydrator\ClassMethodsHydrator;
use Laminas\Form\Annotation\AnnotationBuilder;
/**
 * @ABC\Name("attendee")
 * @ABC\Hydrator("Laminas\Hydrator\ClassMethodsHydrator")
 */
class AttendeeEntity {
    /**
    * @ABC\Name("name")
    * @ABC\Attributes({"type":"text","placeholder":"Name on Ticket"})
    * @ABC\Options({"label":"Name:"})
    * @ABC\Filter({"name":"StringTrim"})
    * @ABC\Filter({"name":"StripTags"})
    */
    protected $name;
}
$form = (new AnnotationBuilder())->createForm('AttendeeEntity');
echo get_class($form);
```

#### Expected behavior

Laminas\Form\Form

**Comment by Slamdunk** [comment]
You are right, we need to prevent `laminas-code:v4` at all for `laminas-form:v2`

**Comment by Slamdunk** [comment]
Closed by #101

```
$ composer require doctrine/annotations

$ composer require laminas/laminas-code
Using version ^4.1 for laminas/laminas-code
./composer.json has been updated
Running composer update laminas/laminas-code
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laminas/laminas-form is locked to version dev-code_4_conflict and an update of this package was not requested.
    - laminas/laminas-form dev-code_4_conflict conflicts with laminas/laminas-code 4.1.0.
    - Root composer.json requires laminas/laminas-code ^4.1 -> satisfiable by laminas/laminas-code[4.1.0].


Installation failed, reverting ./composer.json and ./composer.lock to their original content.

$ composer require laminas/laminas-code:^3
./composer.json has been updated
Running composer update laminas/laminas-code
Loading composer repositories with package information
Updating dependencies
Lock file operations: 1 install, 0 updates, 0 removals
  - Locking laminas/laminas-code (3.5.1)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing laminas/laminas-code (3.5.1): Extracting archive
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Generating autoload files
25 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

$ php annotation_test.php
Laminas\Form\Form
```

**Comment by dbierer** [comment]
Filippo --
   
  
  
   
  
  
   Many many thanks for the quick response.  I can confirm that this works!
   
  
  
   
  
  
   Yours,
   
  
  
   Doug Bierer
   
  
  
   
  
  
   
    On 04/07/2021 2:16 AM Filippo Tessarotto ***@***.***> wrote:
   
   
    
   
   
    
   
   
   You are right, we need to prevent laminas-code:v4 at all for laminas-form:v2
